test/types/schema_helpers.test-d.ts

```diff
@@ -0,0 +1,44 @@
+import type {
+  InferIdType,
+  WithId,
+  // WithoutId,
+  OptionalId
+  // EnhancedOmit
+} from '../../src/mongo_types';
+
+import { expectType, expectNotType, expectAssignable } from 'tsd';
+import { Document, ObjectId } from 'bson';
+
+// InferIdType
+expectType<InferIdType<Document>>(new ObjectId());
+expectType<InferIdType<{ _id: number }>>(1 + 1);
+expectType<InferIdType<{ a: number } | { b: string }>>(new ObjectId());
+expectAssignable<InferIdType<{ _id: number } | { b: string }>>(new ObjectId());
+expectAssignable<InferIdType<{ _id: number } | { b: string }>>(1 + 1);
+
+// WithId
+expectAssignable<WithId<Document>>({ _id: new ObjectId() });
+expectAssignable<WithId<{ a: number }>>({ _id: new ObjectId(), a: 3 });
+expectNotType<WithId<Document>>({ _id: 3 });
+// expectType<WithId<{ _id: number }>>({ _id: 3 });
```

Comment:
lots of commented out expect lines here

Reply:
https://jira.mongodb.org/browse/NODE-3287
Made a ticket about the testing we talked about earlier

Reply:
Looks good, tyvm!